### PR TITLE
Setup k8s deploy to internal cluster

### PR DESCRIPTION
**Pull request by Arthelon**
Addresses #5

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,16 @@
+.PHONY: deploy-image deploy run-migrations
+
+SHORT_SHA=$(shell git rev-parse --short=7 HEAD)
+APP_IMAGE_NAME=gcr.io/oursky-kube/gitlint-bot:${SHORT_SHA}
```
Comment (kiootic) [inline Makefile:4]: Put the image repo in a separate variable. We should tag the image with both:
- `latest`
- the short commit hash

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+deploy:
+	@kubectl -n gitlint-bot apply -f ./k8s-deployment.yaml
+	@kubectl -n gitlint-bot set image deployment/gitlint-bot-production gitlint-bot-production=${APP_IMAGE_NAME}
+
+deploy-image:
+	@docker build --tag ${APP_IMAGE_NAME} .
+	@docker push ${APP_IMAGE_NAME}
+
+run-migrations:
+	@kubectl -n gitlint-bot exec ${POD_NAME} -- npm run migrations
```
Comment (kiootic) [inline Makefile:16]: We should use [Jobs](https://kubernetes.io/docs/concepts/workloads/controllers/job/) for migrations.

File: k8s-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
+apiVersion: v1
```
Comment (kiootic) [inline k8s-deployment.yaml:1]: Please put the k8s manifests in a `deploy` directory at root.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+deploy:
+	@kubectl -n gitlint-bot apply -f ./deploy/k8s-deployment.yaml
+	@kubectl -n gitlint-bot set image deployment/gitlint-bot-production gitlint-bot-production=${APP_IMAGE_SHA}
+
+deploy-image:
+	@docker build -t ${APP_IMAGE_LATEST} -t ${APP_IMAGE_SHA} .
+	@docker push ${APP_IMAGE_LATEST}
+	@docker push ${APP_IMAGE_SHA}
+
+run-migrations:
```
Comment (kiootic) [inline Makefile:17]: It seems this target is not run during the deployment flow. What is the intended usage of this?

File: .github/workflows/nodejs.yml [hunk 5]
```diff
@@ -21,3 +21,28 @@ jobs:
...
       - name: Build TypeScript
         run: npm run build
+  deploy:
+    if: github.event_name == 'push' && github.ref == 'refs/heads/master'
+    needs: test
+    runs-on: ubuntu-latest
+    env:
+      GKE_CLUSTER: ${{ secrets.GKE_CLUSTER }}
+      GCE_ZONE: ${{ secrets.GCE_ZONE }}
+      APP_ID: ${{ secrets.APP_ID }}
```
Comment (kiootic) [inline .github/workflows/nodejs.yml:31]: These (APP_ID, PRIVATE_KEY, WEBHOOK_SECRET) should be set in environment variables of the Pod. You can update the Secret in the internal cluster to include these values.

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,15 @@
+.PHONY: deploy-image deploy run-migrations
```
Comment (kiootic) [inline Makefile:1]: Seems `run-migrations` target is gone?

**Comment by Arthelon** [comment]
Is the `OURSKY_KUBE_GITHUB_RUNNER_KEY` replacing the `GKE_SA_KEY` Github secret?

**Comment by Arthelon** [comment]
What is the `OURSKY_KUBE_GITHUB_RUNNER_KEY` for? Is it a service account key?

**Comment by kiootic** [comment]
Yes, it should be a service account key in JSON format.

**Pull request merged**